integration-cli/docker_cli_build_test.go

```diff
@@ -5251,3 +5251,269 @@ func (s *DockerSuite) TestBuildEmptyStringVolume(c *check.C) {
 	}
 
 }
+
+func (s *DockerSuite) TestBuildBuildTimeEnv(c *check.C) {
+	imgName := "bldenvtest"
+	defer func() { deleteImages(imgName) }()
+	envKey := "foo"
+	envVal := "bar"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-env",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf("FROM busybox\n"+
+		"RUN echo $%s\n"+
+		"CMD echo $%s\n",
+		envKey, envKey))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || !strings.Contains(out, envVal) {
+		if err != nil {
+			c.Fatalf("build failed to complete: %v %v", out, err)
+		}
+		c.Fatalf("failed to access environment variable in output: '%v' "+
+			"expected: '%v'", out, envVal)
+	}
+
+	containerName := "bldenvCont"
+	runCmd := exec.Command(dockerBinary, "run", "--name", containerName, imgName)
+	if out, _, err := runCommandWithOutput(runCmd); out != "\n" || err != nil {
+		c.Fatalf("run produced invalid output: %q, expected %q", out, "")
```

Comment:
No need for the 2nd substitution here since it's just an empty string.
`expected empty string` should be sufficient.